api/types/volume.go

```diff
@@ -10,6 +17,26 @@ type Volume struct {
 	Provider Provider     `json:"provider"`
 }
 
+func NewVolume(name string, size int, provider Provider) (Volume, error) {
+	rand, err := random.GenerateRandomString(11)
+	if err != nil {
+		return Volume{}, fmt.Errorf("failed to generate random string, %w", err)
+	}
+	// not a cryptographically safe identifier, but consistent with execs
+	volID := "vol_" + strings.ToLower(rand)
+
+	return Volume{
+		ID:   volID,
+		Name: name,
```

Comment:
What do you mean they don't have enough entropy? Why do these need to be crypto safe? We using them as identifiers and nowhere for anything auth related. See my comment above for "how" unique they are. 

I'm not sure we should be adding an ID inside a constructor here because in the other parts of the codebase for execs, we use the constructor to create initialize the Exec type whereas here, we use it to add new "identity" to the types. Let's leave it in for now though.

Reply:
entropy doesn't need to just be cryptography related, it just needs to be an id that avoids database collisions on the primary key field, im not sure if these ids are capable of doing that

Reply:
Yeah that's what I was referring to. The crypto part only becomes relevant if you don't want people cracking your uniqueID. That's not a concern here. The algorithm has a collision rate of ~1 in 60! which is tinnnnny.